Amplitude/AMPNSURLSessionNetworkClient.m

```diff
@@ -0,0 +1,60 @@
+//
+//  AMPNSURLSessionNetworkClient.m
+//  Amplitude
+//
+//  Created by Francesco Perrotti Garcia on 10/24/18.
+//  Copyright © 2018 Amplitude. All rights reserved.
+//
+
+#import "AMPNSURLSessionNetworkClient.h"
+#import "AMPARCMacros.h"
+#import "AMPEventUploadRequest.h"
+
+@interface AMPNSURLSessionNetworkClient()
+@property (nonatomic, strong) NSURLSession *session;
+@end
+
+@implementation AMPNSURLSessionNetworkClient
+- (instancetype) init {
+    return [super init];
+}
+
+- (void) uploadEvents: (AMPEventUploadRequest *) uploadRequest using: (NSURLSession *) session completionHandler: (void (^)(NSData *  data, NSURLResponse *response, NSError *error)) completionHandler {
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:uploadRequest.url];
+    [request setTimeoutInterval:60.0];
+
+    NSString *apiVersionString = [[NSNumber numberWithInt:uploadRequest.apiVersion] stringValue];
+
+    NSMutableData *postData = [[NSMutableData alloc] init];
+    [postData appendData:[@"v=" dataUsingEncoding:NSUTF8StringEncoding]];
```

Comment:
It feels just a little bit weird to me that this logic is inside the NetworkClient. This is our event server API, and I feel like regardless of what network protocol or logic you use, the data format will be the same? Maybe the input to `uploadEvents` should just be a `NSMutableData *postData` that is constructed by the parent caller?

Is your custom client going to be using repeating all of this logic or do you have a different format for your own endpoint?

Reply:
We were using JSON (instead of Form URL encoded) in our server, but we're refactoring it to use Form URL Encoding so we can simplify this PR.

Reply:
After discussing the matter internally we arrived to the conclusion that it made more sense to maintain the `AMPEventUploadRequest` as a wrapper of a request instead of just sending data and leaving the HTTP method, headers and request URL to be implicitly created/inferred by the network clients. WDYT about this?